test/scripts/get_ext_ip.sh

```
@@ -1,8 +1,21 @@
+    else
+        result=$(ip route get 1.1.1.1 2>/dev/null | grep -oP 'src \K\S+' || true)
+    fi
+    echo "$result"
 else
     # MacOS does not have ip, so we use route and ifconfig instead
-    INTERFACE=$(route get 1.1.1.1 | grep interface | awk '{print $2}')
-    ifconfig | grep $INTERFACE -A 10 | grep "inet " | grep -Fv 127.0.0.1 | awk '{print $2}' | head -n 1
+    if [[ "${IPV6_ONLY:-false}" == "true" ]]; then
+        # IPv6-only for MacOS
+        INTERFACE=$(route get -inet6 2001:4860:4860::8888 2>/dev/null | grep interface | awk '{print $2}')
+        ifconfig | grep $INTERFACE -A 10 | grep "inet6 " | grep -Fv "::1" | grep -Fv "fe80:" | awk '{print $2}' | head -n 1
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)